### PR TITLE
 Create payment summary page

### PR DESCRIPTION
**Pull request by e-wai**
## Notion ticket link

[Registrant Experience: Payment Summary Page](https://www.notion.so/uwblueprintexecs/Registrant-Experience-Payment-Summary-Page-cb1d01aecd384edfa55776b6e188a866)

Also fixes some tablet/mobile styling on the waiver page

## Steps to test
1. It's the current page for the "review" step, since nothing else exists there atm. I'm working on some stepper fixes that will make the flow a bit clearer, but the component itself is done 

Screen Shot 2022-12-14 at 7 28 50 PM

## What should reviewers focus on?
* Check mobile styling, scrolling behaviour

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/RegistrantExperience/ReviewRegistration/index.tsx [hunk 1]
```diff
@@ -1,26 +1,64 @@
 import React from "react";
-import { Box, Checkbox, Text } from "@chakra-ui/react";
+import { Box } from "@chakra-ui/react";
+import PaymentSummary from "./PaymentSummary";
+import { CartItem } from "../../../../types/RegistrationTypes";
+import ReviewInformation from "./ReviewInformation";
```
Comment (e-wai) [inline frontend/src/components/pages/RegistrantExperience/ReviewRegistration/index.tsx:5]: This is just used to setup the component and to have the linter be mad about it being dead code, I assume @twilkhoo's ticket will build out this component

File: frontend/src/types/RegistrationTypes.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export type CartItem = {
+  name: string;
+  campers: number;
+  totalPrice: number;
+  details?: string;
```
Comment (dhmi19) [inline frontend/src/types/RegistrationTypes.ts:5]: what is the details field for and why is it optional?

File: frontend/src/types/RegistrationTypes.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export type CartItem = {
```
Comment (dhmi19) [inline frontend/src/types/RegistrationTypes.ts:1]: Why do we need a CartItem type? I thought we could use the array of camper objects to determine these fields? Or was this just a temporary work-around?

**Comment by dhmi19** [comment]
I think we need to add some more padding for the mobile view:
image

**Comment by dhmi19** [comment]
likewise for the tablet view:
image

Also, the scrolling seems scroll the whole page not the table. Would it be possible to add some more dummy data so we have enough overflow to test the table in tablet and mobile?

File: frontend/src/components/pages/RegistrantExperience/RegistrationFooter.tsx [hunk 4]
```diff
@@ -34,6 +34,7 @@ const RegistrationFooter = ({
       padding="20px"
```
Comment (dhmi19) [inline frontend/src/components/pages/RegistrantExperience/RegistrationFooter.tsx:34]: I think the designs say that the button on the review cart page should be "Go to Checkout" not "pay". Could you fix that up in this PR? On the review info step (what tejas is working on, also part of step 4) the CTA is next, but when it comes to the cart review page, the CTA is "go to checkout"

**Review by dhmi19 — COMMENTED** [review]
I think the ui looks mostly ok. Just left some minor comments about responsiveness. I am guessing the state you used is temporary for now, if not, I have some questions lol

File: frontend/src/types/RegistrationTypes.ts (hunk 2 above)
Comment (e-wai) [reply frontend/src/types/RegistrationTypes.ts:5]: The "150 minutes" field, but yeah I might've forgotten to render it, will add

Screen Shot 2022-12-14 at 11 19 43 PM

File: frontend/src/types/RegistrationTypes.ts (hunk 3 above)
Comment (e-wai) [reply frontend/src/types/RegistrationTypes.ts:1]: Yeah I'll prob map directly eventually, this was a nice in between for dummy props

**Comment by e-wai** [comment]
@dhmi19 The padding looks ok on my screen?

https://user-images.githubusercontent.com/40974372/207781804-e0d81249-8132-4ede-a111-25a1f4bb5502.mov

**Comment by dhmi19** [comment]
> @dhmi19 The padding looks ok on my screen?
> 
>  Screen.Recording.2022-12-15.at.12.42.16.AM.mov

I think you need to refresh the page when you change the size of the screen

**Review by dhmi19 — APPROVED** [review]
lgtm :))

**Pull request merged**